components/ServerForm.tsx

```diff
@@ -55,10 +85,58 @@ export default function ServerForm({
         inputMode="url"
         keyboardType="url"
         autoCapitalize="none"
-        onSubmitEditing={validateAndSaveURL}
-        returnKeyType="go"
+        onSubmitEditing={() =>
+          basicAuth.basicAuthRequired
+            ? usernameRef.current.focus()
+            : validateAndSaveURL()
+        }
+        returnKeyType={basicAuth.basicAuthRequired ? "next" : "go"}
         autoCorrect={false}
       />
+      <CheckBox
+        checked={basicAuth.basicAuthRequired}
+        onChange={(nextValue) => setBasicAuthRequired(nextValue)}
+      >
+        Anmeldung erforderlich
+      </CheckBox>
+      {basicAuth.basicAuthRequired && (
+        <Input
+          style={{ marginTop: 5 }}
+          size="large"
+          status={validateUsername() ? "basic" : "danger"}
```

Comment:
Having it initially read is irritating. I'd skip validation completely on both fields and always do basic status without label.